task5/src/main/java/com/clouway/adapter/persistence/PersistentFundsRepository.java

```diff
@@ -89,4 +95,65 @@ public boolean withdraw(Double amount, String email) {
     }
     return succesfull;
   }
+
+  public void updateHistory(Long date, String email, String operation, Double amount) {
+    Connection connection = connectionProvider.get();
+    PreparedStatement statement = null;
+    try {
+      statement = connection.prepareStatement("INSERT INTO transactions (date, email, operation, amount) VALUES (?,?,?,?)");
+      statement.setLong(1, date);
+      statement.setString(2, email);
+      statement.setString(3, operation);
+      statement.setDouble(4, amount);
+      statement.execute();
+    } catch (SQLException e) {
+      e.printStackTrace();
+    } finally {
+      try {
+        statement.close();
+      } catch (SQLException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  public List<Transaction> getHistory(Integer limit, Integer offset) {
+    Connection connection = connectionProvider.get();
+    PreparedStatement statement = null;
+    List<Transaction> transactions = new ArrayList<Transaction>();
+    try {
+      statement = connection.prepareStatement("SELECT * FROM transactions LIMIT " + limit + " OFFSET " + offset + "");
+      ResultSet resultSet = statement.executeQuery();
+      while (resultSet.next()) {
+        String date = DateFormat.getDateTimeInstance().format(new Date(resultSet.getLong(2)));
```

Comment:
When you are using dates I told you to avoid working with date like string. Use Long or Date or there is DateTime that is part of Joda Time library. I allready explaind you why. Leave it like this because no time